lib/base-service.ts

```diff
@@ -141,7 +141,9 @@ export class BaseService {
    * @param {string} url The base URL for the service.
    */
   public setServiceUrl(url: string): void {
-    this.baseOptions.serviceUrl = url;
+    if (url) {
```

Comment:
Given a parameter type of just `string`, this if statement would be superfluous. Should the type be changed to `url?: string` or the if statement removed?

Reply:
Yeah I think that's a fair point, the `if` statement probably doesn't need to be there. Or should be stricter and ensure that it's actually a string. In the JS case (which we think is most of our users), the type or requiredness isn't actually enforced. And `stripTrailingSlash` uses the `.replace()` method so it will throw a TypeError if `url` isn't a string.

I agree that the `if` statement as is isn't really that useful